Sources/TuistCache/Utilities/CacheXCFrameworkBuilder.swift

```diff
@@ -133,19 +136,15 @@ public final class CacheXCFrameworkBuilder: CacheArtifactBuilding {
                 ))
             }
 
-            if let buildTargets = buildTargets {
-                for target in buildTargets {
-                    if target.name == productName {
-                        let isSupported = self.isMacCatalystSupported(projectPath: target.projectPath, projectName: target.name)
-                        if isSupported {
-                            if let macCatalystArchivePath = macCatalystArchivePath {
-                                frameworkpaths.append(self.frameworkPath(
-                                    fromArchivePath: macCatalystArchivePath,
-                                    productName: productName
-                                ))
-                            }
-                        }
-                    }
+            for target in buildTargets where target.name == productName && self.isMacCatalystSupported(
+                projectPath: target.projectPath,
+                projectName: target.name
+            ) {
```

Comment:
nit: I would use filter here
```suggestion
            buildTargets
                .filter { $0.name == productName }
                .filter { self.isMacCatalystSupported(projectPath: $0.projectPath, projectName: $0.name) }
                .forEach {
```

Reply:
This approach looks more readable. fixed !